system/doc/reference_manual/processes.xml

```diff
@@ -109,23 +185,508 @@ spawn(Module, Name, Args) -> pid()
   </section>
 
   <section>
-    <title>Message Sending</title>
-    <p>Processes communicate by sending and receiving messages.
-      Messages are sent by using
-      the <seeguide marker="expressions#send">send operator !</seeguide>
-      and received by calling
-      <seeguide marker="expressions#receive">receive</seeguide>.</p>
-    <p>Message sending is asynchronous and safe, the message is
-      guaranteed to eventually reach the recipient, provided that
-      the recipient exists.</p>
+    <title>Signals</title>
+    <p>
+      <marker id="message-sending"/>
+      All communication between Erlang processes and Erlang ports is done by
+      sending and receiving asynchronous signals. The most common signals are
+      Erlang message signals. A message signal can be sent using the
+      <seeguide marker="expressions#send">send operator <c>!</c></seeguide>.
+      A received message can be fetched from the message queue by the
+      receiving process using the
+      <seeguide marker="expressions#receive"><c>receive</c></seeguide>
+      expression.
+    </p>
+    <section>
+      <title>Sending Signals</title>
+
+      <p>
+	Besides message signals there are a lot of other signals that can be
+	sent. The set of signals used may vary. A signal may be replaced by
```

Comment:
"Vary" in which way? Over time? Between releases? Something else? I think you should clarify or remove this sentence.

"may be replaced by": Be replaced by whom? By the runtime system? The user?

Reply:
Yes, this should be clarified.

This was what I was thinking about when I wrote this:
* A signal may be replaced by another type of signal in a patch. For example, a reply signal as a response to a `gc_request` signal is currently always a message signal, but can in the synchronous case be replaced by a non-message signal specially crafted for this reply. Such a replacement if fully compatible and can be introduced in any patch.
* A signal type may change at runtime depending on the state of the system. For example, a reply signal to an `alive_request` may currently be a message signal in some cases, but in other cases not, depending on the states of the involved processes.

However, the signals are are still `gc_reply` and `alive_reply` regardless of whether they are realized using message signals or not. I'll do some rewriting here...